ipfssync/subpub/subpub.go

```diff
@@ -83,55 +76,69 @@ type Message struct {
 // The groupKey is a secret shared among the PubSub participants. Only those with the key will be able to join.
 // If private enabled, a libp2p private network is created using the groupKey as shared secret (experimental).
 // If private enabled the default bootnodes will not work.
-func NewSubPub(key ecdsa.PrivateKey, groupKey []byte, port int32, private bool) *SubPub {
+func NewSubPub(hexKey string, groupKey []byte, port int32, private bool) *SubPub {
 	ps := new(SubPub)
-	ps.Key = key
+
+	s := ethereum.NewSignKeys()
+	if err := s.AddHexKey(hexKey); err != nil {
+		log.Fatalf("cannot import privkey %s: %s", hexKey, err)
+	}
+	ps.Key = s.Private
+
 	if len(groupKey) < 4 {
 		panic("subpub group key is too small; 4 bytes at minimum")
 	}
 	copy(ps.GroupKey[:], ethereum.HashRaw(groupKey)[:32])
 	ps.Topic = fmt.Sprintf("%x", ethereum.HashRaw([]byte("topic"+string(groupKey))))
-	ps.PubKey = hex.EncodeToString(eth.CompressPubkey(&key.PublicKey))
-	ps.privKey = hex.EncodeToString(key.D.Bytes())
-	ps.BroadcastWriter = make(chan []byte)
+	ps.PubKey = hex.EncodeToString(eth.CompressPubkey(&ps.Key.PublicKey))
+	ps.privKey = hex.EncodeToString(ps.Key.D.Bytes())
 	ps.Reader = make(chan *Message)
```

Comment:
if you want to avoid blocking on the send line to Reader where I suggested to remove the goroutine, you could add a bit of buffering here, so we can e.g. read and buffer ten messages in Reader before making the network stream block.